### ISSUE TITLE
Multiple compressedFile.extract() return wrong content

### ISSUE THREAD

**Issue by Lenqth** [nika-begiashvili/libarchivejs#28]
- Problem
Concurrent compressedFile.extract() execution breaks correspondence of files with contents.
Because when message cames from webworker, always called the last element of _callbacks in current implementation. but multiple extractSingleFile calling cause _callbacks to stack up and return contents in wrong order.

 - How to reproduce

Modify test/files/test-single.html :
```javascript
@@ -65,7 +65,7 @@
                     const file = e.currentTarget.files[0];
                     const archive = await Archive.open(file);
                     const files =  await archive.getFilesArray();
-                    fileObj = await files[0].file.extract();
+                    fileObj = (await Promise.all(files.map(f =>f.file.extract()) ))[0];
                 }catch(err){
                     console.error(err);
                 }finally{
```
and `npm run test`

 - How to fix
Attach message id to message for communicate with WebWorker and response are routed by message id.

**Comment by janpaul123** [comment]
I ran into this too. It's a race condition, because the returning `postMessage` calls are not guaranteed to fire in order.

**Comment by nika-begiashvili** [comment]
Fixed in #64
